auth/src/main/java/com/firebase/ui/auth/ui/email/RegisterEmailActivity.java

```diff
@@ -73,24 +87,43 @@ protected void onCreate(Bundle savedInstanceState) {
 
         mSaveSmartLock = mActivityHelper.getSaveSmartLockInstance();
 
-        mEmailEditText = (EditText) findViewById(R.id.email);
-        mNameEditText = (EditText) findViewById(R.id.name);
-        mPasswordEditText = (EditText) findViewById(R.id.password);
-
         mPasswordFieldValidator = new PasswordFieldValidator(
                 (TextInputLayout) findViewById(R.id.password_layout),
                 getResources().getInteger(R.integer.min_password_length));
         mNameValidator = new RequiredFieldValidator((TextInputLayout) findViewById(R.id.name_layout));
         mEmailFieldValidator = new EmailFieldValidator((TextInputLayout) findViewById(R.id.email_layout));
 
+        mEmailEditText = (EditText) findViewById(R.id.email);
+        mNameEditText = (EditText) findViewById(R.id.name);
+        mPasswordEditText = (EditText) findViewById(R.id.password);
+
+        mEmailEditText.setOnFocusChangeListener(this);
+        mNameEditText.setOnFocusChangeListener(this);
+        mPasswordEditText.setOnFocusChangeListener(this);
+        findViewById(R.id.button_create).setOnClickListener(this);
+
+        setUpTermsOfService();
+
+        // Activity rotated
+        if (savedInstanceState != null) {
+            mPrevEmail = savedInstanceState.getString(PREV_EMAIL);
+            return;
+        }
+
         String email = getIntent().getStringExtra(ExtraConstants.EXTRA_EMAIL);
-        if (email != null) {
+        if (!TextUtils.isEmpty(email)) {
             mEmailEditText.setText(email);
-            mEmailEditText.setEnabled(false);
+            mNameEditText.requestFocus();
+            return;
         }
-        setUpTermsOfService();
-        Button createButton = (Button) findViewById(R.id.button_create);
-        createButton.setOnClickListener(this);
+
+        showEmailAutoCompleteHint();
```

Comment:
Technically the email hint is provided by SmartLock, so this would make the API somewhat dishonest.  Although I agree with you that enabling the hint for everyone may be a superior UX, I want to respect the intent of the developer when they say "disable smartlock".